google/ads/google_ads/client.py

```diff
@@ -346,6 +347,21 @@ def __init__(self, developer_token, login_customer_id):
             ('login-customer-id', login_customer_id) if login_customer_id
             else None)
 
+    def _update_client_call_details_metadata(
+            self, client_call_details, metadata):
+        class _ClientCallDetails(
```

Comment:
Google style guide recommends avoiding inner classes - I would move this definition outside of `class MetadataInterceptor`

Reply:
Awesome, that's great feedback. That would also make it easier to access in the tests if we decided to do so.